src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,45 @@
+import {Directive, ElementRef, EventEmitter, HostListener, Output} from '@angular/core';
+import {LoadDirection} from './load-direction.type';
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class ScrollWithLoadingDirective {
+    @HostListener('scroll')
+    onScroll() {
+        const {nativeElement} = this.elementRef;
```

Comment:
А почему бы не деструктуризировать `nativeElement`? Было бы намного удобней как мне кажется
```suggestion
        const {...} = this.elementRef.nativeElement;
```

А в целом, есть ли необходимость поучать элемент через конструктор, почему бы его не взять из `target` события? Мне кажется, получилось бы проще

Reply:
да, проще.
Я подумал, что принципиальной разницы нет, поэтому взял из конструктора.